src/config.cpp

```diff
@@ -103,99 +189,170 @@ const Config& config()
-        ("metric-leader", po::value(&config.metric_leader)->default_value(QUOTE(DEFAULT_METRIC_LEADER)),
-             "name of leading perf event")
-        ("metric-count", po::value(&metric_count),
-             "# of events to elapse by metric leader before reading metric buffer")
-        ("metric-frequency", po::value(&metric_frequency),
-             "metric buffer reads per second")
+            "Show this help message.")
+        ("version",
+            "Print version information.")
+        ("output-trace,o",
+            po::value(&config.trace_path)
+                ->value_name("PATH"),
+            "Output trace directory.")
+        ("quiet,q",
+            po::bool_switch(&config.quiet),
+            "Suppress output.")
+        ("verbose,v",
+            po::value(&verbosity)
+                ->zero_tokens(),
+            "Verbose output (specify multiple times to get increasingly more verbose output).")
+        ("mmap-pages,m",
+            po::value(&config.mmap_pages)
+                ->value_name("PAGES")
+                ->default_value(16),
+            "Number of pages to be used by internal buffers.")
         ("clockid,k",
-             po::value(&requested_clock_name)->default_value("monotonic-raw"),
-             "clock used for perf timestamps (see --list-clockids for supported arguments)")
-#ifdef HAVE_X86_ADAPT // I am going to burn in hell for this
-        ("x86-adapt-cpu-knob,x", po::value(&config.x86_adapt_cpu_knobs),
-             "add x86_adapt knobs as recordings. Append #accumulated_last for semantics.")
-#endif
-        ("disassemble", po::bool_switch(&disassemble),
-             "enable augmentation of samples with instructions (default if supported)")
-        ("no-disassemble", po::bool_switch(&no_disassemble),
-             "disable augmentation of samples with instructions")
-        ("kernel", po::bool_switch(&kernel),
-             "include events happening in kernel space (default)")
-        ("no-kernel", po::bool_switch(&no_kernel),
-             "exclude events happening in kernel space")
-        ("list-clockids", po::bool_switch(&list_clockids)->default_value(false),
-            "list all available clockids")
-        ("list-events", po::bool_switch(&list_events)->default_value(false),
-            "list all available events")
-        ("command", po::value(&config.command));
+            po::value(&requested_clock_name)
+                ->value_name("CLOCKID")
+                ->default_value("monotonic-raw"),
+            "Clock used for perf timestamps (see --list-clockids for supported arguments).")
+        ("list-clockids",
+            po::bool_switch(&list_clockids)
+                ->default_value(false),
+            "List all available clockids.")
+        ("list-events",
+            po::value(&list_event_category)
+                ->value_name("CATEGORY")
+                ->implicit_value("all"),
```

Comment:
The help message generated from this looks different than other options.
Considering that I'd like to split this anyways, just turn this into a bool_switch and print predefined and pmu events.
`--list-tracepoints` than should print the list of tracepoints.